src/features/DebugSession.ts

```diff
@@ -80,6 +80,12 @@ export class DebugSessionFeature implements IFeature {
                     }
                 }
             }
+            else if (config.script) {
+                // In this case, the user has explicitly defined a script path
+                // so make sure to set the cwd to that path if the cwd wasn't
+                // explicitly set
+                config.cwd = config.cwd || config.script;
```

Comment:
Sounds good.  Then for the unsaved file, the cwd will come through as undefined, then PSES will temporarily set it to `untitled:Untitled-N` and then set it back to Env.CurrentDir or  `AppContext.BaseDirectory` on CoreCLR.  We might want to cleanup the handling of this on the PSES later but should be OK for now.  I'll submit my PR soon for you to check out.